src/services/CleanupService.ts

```diff
@@ -754,42 +874,44 @@ export const cleanCookiesOperation = async (
       );
     });
 
-    const markedForLocalStorageDeletion = isSafeToCleanObjects.filter((obj) => {
-      const r = filterLocalstorage(obj, debug);
-      cadLog(
-        {
-          msg: `CleanupService.filterLocalstorage: returned ${r} for ${obj.cookie.hostname}`,
-        },
-        debug,
-      );
-      return r;
-    });
-
-    // Side effects
-    allLocalstorageToClean = [
-      ...allLocalstorageToClean,
-      ...markedForLocalStorageDeletion,
-    ];
-
-    // Clean other browsingdata.  Pass in Domain for specific cleanup for LocalStorage.
-    const domainsToClean = allLocalstorageToClean
-      .map((obj) => obj.cookie.domain)
-      .filter((domain) => domain.trim() !== '');
-
-    try {
-      await otherBrowsingDataCleanup(state, domainsToClean);
-    } catch (e) {
-      cadLog(
-        {
-          type: 'error',
-          x: { e, domainsToClean },
-        },
-        true,
-      );
-      // if it reaches this point then cookies were deleted, so don't return undefined
-      throwErrorNotification(e);
+    // Handle all other browsingData cleanups.
+    const storeResults = await otherBrowsingDataCleanup(
+      state,
+      isSafeToCleanObjects,
+    );
+    if (storeResults.cache) {
```

Comment:
Probably be best to make a helper function here so you don't have to declare `setOfDeletedDomainX` above and repeat the logic here many times.

Like the input would be `storeResults.cache` =>`setOfDeletedDomainCaches`

Another thing is that you can do `new Set(array)`

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Set/Set

Reply:
So what you're thinking is to create an array of objects returned from `otherBrowsingDataCleanup`, then outside the container loops, parse and combine the domains from each site data type from all containers and use Set once to remove dups?

Reply:
Implemented in commit ac90490